design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,455 @@
+It will then fetch all the K8s secrets update them with the Conjur secrets they require and terminate upon completion.
+
+#### How does a Job installed via Helm answer the requirements?
+
+Running as a **Job** allows separation from the applications' deployment and serve multiple applications at once. 
+
+Because for this Milestone we are concerned with updating K8s Secrets once at intial spin up, a **Job** is the most native solution. It will terminate upon task completion and not waste customer's resources.
+
+Kubernetes Helm is a tool that streamlines and organizes the management of Kubernetes installation and deployment processes. When we use Helm for deployment and lifecycle management, we can provide our customer's a one-click solution even though the way at which we deploy changes.
+
+#### What drawbacks does this solution have?
+
+**Job** is not an existing application identity granularity.
+
+ As a result, one can define the host representing the Secrets Provider in Conjur policy using only **namespace** and/or **service account** granularities.
+
+To handle the missing **Job** granularity there are 2 options:
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Keep using only **namespace** and/or **service account** granularities. | \- Since Secrets Provider runs as a separate entity, we suggest creating a dedicated service account for Secrets Provider.<br />That way, the **service account** granularity serves as an exact match and **Job** granularity is redundant<br />- No code changes | \- Service account might be reused, which allows other apps to authenticate using Secrets Provider host<br />- Break application identity convention | Free              |
+| 2    | Add support for **Job** granularity in Conjur                | Allows specific identification of the Secrets Provider **Job**, which enhances security | \- Costly; requires code changes, verify compatibility, tests, docs<br />- Redundant if **service account** serves only the Secrets Provider | 10 days           |
+
+*Decision*: Solution #1, use **namespace** and **service account** to define the Secrets Provider host in Conjur policy. We would recommend to the customer to create a dedicated service account for Secrets Provider.
+
+#### Customer experience
+
+Deployment for the Secrets Provider will be done using Helm. That way, if the 
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined
+
+2. Create `values.yaml` that will contain the following parameters and their defaults:
```

Comment:
is `values.yaml` a known best practice for helm? If not, how about `secrets-provider-values.yaml`?

Reply:
From what I understood, `values.yaml` is the standard defaults file
@oburstein-hub 

Now that I think of it, could a customer ideally supply us with multiple files that override the values.yaml and load them in like so:
```
`helm install conjur -f custom-values.yaml custom-values2.yaml custom-values3.yaml https://github.com/cyberark/secrets-provider-for-k8s/helm/secrets-provider-1.0.0.tgz`
```

Reply:
right

Reply:
is this a question or a statement?